fortran/CMakeLists.txt

```diff
@@ -103,37 +103,42 @@ endif()
-  target_link_libraries (example_${name}_fortran_exe PRIVATE primaf)
-  target_include_directories (example_${name}_fortran_exe PRIVATE ${CMAKE_BINARY_DIR}/fortran)
-  set_target_properties(example_${name}_fortran_exe PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/examples/${name}/mod)
+macro (prima_add_f_test name number)
+  add_executable (example_${name}_fortran_${number}_exe EXCLUDE_FROM_ALL examples/${name}/${name}_example_${number}.f90)
+  target_link_libraries (example_${name}_fortran_${number}_exe PRIVATE primaf)
+  target_include_directories (example_${name}_fortran_${number}_exe PRIVATE ${CMAKE_BINARY_DIR}/fortran)
+  set_target_properties(example_${name}_fortran_${number}_exe PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/examples/${name}_${number}/mod)
   if (WIN32)
-    set_target_properties(example_${name}_fortran_exe PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+    set_target_properties(example_${name}_fortran_${number}_exe PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
   endif()
   if (PRIMA_ENABLE_EXAMPLES)
-    set_target_properties (example_${name}_fortran_exe PROPERTIES EXCLUDE_FROM_ALL FALSE)
+    set_target_properties (example_${name}_fortran_${number}_exe PROPERTIES EXCLUDE_FROM_ALL FALSE)
   endif ()
 
   # Outside of CI we don't want to force people to run examples with gdb, so we test the executables by themselves.
   # We want these to run in CI as well, because sometimes running with gdb masks an error, so we set them up
   # before we set up the examples for CI
-  add_test (NAME example_${name}_fortran COMMAND example_${name}_fortran_exe)
+  add_test (NAME example_${name}_fortran_${number} COMMAND example_${name}_fortran_${number}_exe)
 
   # Within CI, we'd like to run with gdb so that if there's a segfault the logs will have a stacktrace we can use to investigate.
   # Of course this can be run locally as well if you define CI in your environment.
   if(NOT APPLE AND UNIX AND DEFINED ENV{CI})  # Apple security policy will not allow running gdb in CI
-    add_test (NAME example_${name}_fortran_with_gdb COMMAND gdb -batch --command=${CMAKE_BINARY_DIR}/cmdfile.gdb example_${name}_fortran_exe)
+    add_test (NAME example_${name}_fortran_${number}_with_gdb COMMAND gdb -batch --command=${CMAKE_BINARY_DIR}/cmdfile.gdb example_${name}_fortran_${number}_exe)
   elseif(WIN32 AND DEFINED ENV{CI})
     # For Windows we need to provide the full path to the executable since it is installed to a different directory
-    add_test (NAME example_${name}_fortran_with_gdb COMMAND gdb -batch --command=${CMAKE_BINARY_DIR}/cmdfile.gdb ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/example_${name}_fortran_exe.exe)
+    add_test (NAME example_${name}_fortran_${number}_with_gdb COMMAND gdb -batch --command=${CMAKE_BINARY_DIR}/cmdfile.gdb ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/example_${name}_fortran_${number}_exe.exe)
```

Comment:
- CMAKE_INSTALL_BINDIR is relative to the install dir and should not be used here,
instead the generator expression "$<TARGET_FILE:example_..._exe>" can be used,
this should allow to also get rid of the if/else
- no need to run the test twice, with and without gdb, instead this could be an option that decides if test are run with gdb or not (maybe useful also locally), in a general manner we dont want to tie too much things related to the environment (ci or not) in the cmake infra. also we could want to check if gdb is available first with find_program

Reply:
CMAKE_INSTALL_BINDIR was not introduced in this PR and it would be inappropriate to change it here since this PR is about the examples. If you'd like to make a separate PR to change it please feel free and encouraged to do so.

We run the test both with and without gdb because we have seen a case where there is a segfault when not running with gdb, and no segfault under gdb, so we run both tests to catch as many errors as possible.